gufe/tests/test_protocoldag.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of gufe and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+import os
+import pathlib
+import pytest
+from openff.units import unit
+
+import gufe
+from gufe.protocols import execute_DAG
+
+
+class WriterUnit(gufe.ProtocolUnit):
+    @staticmethod
+    def _execute(ctx, **inputs):
+        my_id = inputs['identity']
+
+        with open(os.path.join(ctx.shared, f'unit_{my_id}_shared.txt'), 'w') as out:
+            out.write(f'unit {my_id} existed!\n')
+        with open(os.path.join(ctx.scratch, f'unit_{my_id}_scratch.txt'), 'w') as out:
+            out.write(f'unit {my_id} was here\n')
+
+        return {
+            'log': 'finished',
+        }
```

Comment:
I think this is where @dwhswenson would say these files need tagging in the output to exist